### ISSUE TITLE
Custom errors doesn't not work in React Native

### ISSUE THREAD

**Issue by jakubkoci** [hyperledger/aries-framework-javascript#281]
I found out that `make-error-cause` we're using for custom framework errors doesn't work in React Native. It has dependency on `inspect` function from `util` Node.js module.

I'm sorry, that's my fault. I recommended to use the library because I thought we're using it in another React Native project. However, I found out we're using just `make-error`. 

I'm thinking about how to fix it. One way could be to use `make-error` and just store `cause` error as a property of custom error. Then we wouldn't lose original `error`. The problem could be a missing stack trace.

**Comment by TimoGlastra** [comment]
Ouch. I also didn't take the extra effort to validate it works in RN...

Does `make-error` still provide benefit? The nice thing of `make-error-cause` is that it handled the `cause` part. But if we need to do that ourselves, we can just use the native `Error` rigth?

**Comment by jakubkoci** [comment]
It's a question how we would implement that. It seems that `make-error` provides these [features](https://www.npmjs.com/package/make-error#features). 

Maybe TypeScript helps with some of them already. I found this https://stackoverflow.com/questions/41102060/typescript-extending-error-class.

I'm not sure about preserving stack trace. The `Error.captureStackTrace();` is not available in RN. 

I've made it compile with my fork of `make-error-cause` using [util-inspect](https://github.com/Automattic/util-inspect). But haven't tested yet, I have some dependency injection error now :)

**Comment by jakubkoci** [comment]
So this is the output log with my updates to `make-error-cause`. It seems it works:
```
 LOG  ====================
 LOG  inner error
 ERROR  [SyntaxError: JSON Parse error: Unexpected EOF]
 LOG  outer error
 ERROR  [AriesFrameworkError: Custom Aries Framework message]
 ERROR  [SyntaxError: JSON Parse error: Unexpected EOF]
 LOG  captureStackTrace@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:134672:32
BaseError@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:134714:22
BaseError@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:133288:30
construct@[native code]
_createSuperInternal@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:133221:406
AriesFrameworkError@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:133241:25
initAgent$
tryCatch@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:7706:23
invoke@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:7879:32
tryCatch@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:7706:23
invoke@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:7779:30
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:7789:21
tryCallOne@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:28575:16
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:28676:27
_callTimer@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:29092:17
_callImmediatesPass@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:29131:17
callImmediates@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:29349:33
__callImmediates@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:3230:35
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:3008:34
__guard@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:3213:15
flushedQueue@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.ariesmobileagent&modulesOnly=false&runModule=true:3007:21
flushedQueue@[native code]
invokeCallbackAndReturnFlushedQueue@[native code]
```

This is the code:
```js
  try {
    try {
      JSON.parse(''); // throw some error
    } catch (error) {
      console.log('inner error');
      console.error(error);
      throw new AriesFrameworkError('Custom Aries Framework message', {
        cause: error,
      });
    }
  } catch (outerError) {
    console.log('outer error');
    console.error(outerError);
    console.error(outerError.cause);
    console.log(outerError.stack);
  }
```

**Comment by jakubkoci** [comment]
Fixed by #285
